## Homework 4

Modern Stellar Astrophysics <br>
Melissa Ness <br>
December 11th 2018 <br>
Yasmeen Asali

In [3]:
import numpy as np
import astropy.units as u

### Problem 3

### Problem 4

**Part a)**

The equation for outward radiation force is given by:

$$ F_{rad} = \frac{\kappa L}{4 \pi R^{2}c} $$

And gravitational force per unit mass at the surface where $r=R$ is given by:

$$ F_{grav} = \frac{G M}{R^{2}} $$

The limit where the blob is ejected from the star occurs when the outward radiation force becomes strong enough to overcome the gravitational attraction that keeps the blob attached to the surface. Mathematically, it means $ F_{rad} >  F_{grav}$. 

$$ \frac{G M}{R^{2}} < \frac{\kappa L}{4 \pi R^{2}c} $$

$$ {G M} < \frac{\kappa L}{4 \pi c} $$

$$ \frac{M}{L} < \frac{\kappa}{4 \pi c G} $$

**Part b)**

The max luminosity is 

$$ L = \frac{4 \pi c G M}{\kappa} $$

In [8]:
c = 3e10 * (u.cm / u.s)
G = 6.67 * 10**(-8) * ((u.cm)**3 / (u.g * (u.s)**2))
k = 0.3 * ((u.cm)**2 / u.g)

lum_const = (4*np.pi*c*G)/k
print(lum_const)

83817.69199777568 cm2 / s3


So the constants give a value of 83817, which we can multiply by mass to give:

$$ L_{max} = 83817 \frac{M}{M_{\odot}} \frac{erg}{s} $$

Note that the units work out because $ erg = \frac{g cm^{2}}{s^{2}} $. 

This is the called the Eddington luminosity. 

** Part c) **



### Problem 5

** Part a) **

The rate of mass loss will increase as the luminosity goes up, which intuitively makes sense since radiation can drive stellar winds that drive mass away from the star. As the rate of mass loss increases, the mass of the star gets smaller faster. 

** Part b) **

First we calculate radius:

$$ L = 4 \pi R^{2} \sigma T^{4} $$

$$ R = (\frac{L}{4 \pi \sigma T^{4}})^{1/2} $$

In [11]:
L = 7000 * (3.826 * 10**33 * (u.erg/u.s))
T = 3000 * u.K
sigma = 5.6704 * 10**(-5) * (u.erg / (u.cm**2 * u.s * u.K**4))

R = np.sqrt(L / (4 * np.pi * sigma * T**4))
print(R)

21541067096026.137 cm


Next we have to calculate the surface gravity using this formula: 

$$ g = \frac{GM}{R^{2}} $$

In [13]:
M_sol = 1.9891 * 10**33 * u.g
g = (G*M_sol) / (R**2)
print(g)

0.28592229715002837 cm / s2


The last step is plugging in our values for g and R into the mass loss formula to solve for the mass loss rate of a solar mass star. Since $\eta$ is near unity, we can let it equal one and exclude it from calculations.

In [15]:
g_sol = 2.74 * 10**4 * (u.cm / u.s**2)
L_over_Lsol = 7000
g_over_gsol = g / g_sol
R_over_Rsol = R / (6.955 * 10**10 * u.cm)

mass_loss = (- 4 * 10**(-13) * (L_over_Lsol)) / ((g_over_gsol) * (R_over_Rsol))
print(mass_loss) 

-8.663442367505272e-07


So our final mass loss rate is $ \boxed{ -8.66 * 10^{-7} M_{\odot} yr^{-1} } $. 

** Part c) **

Expressing $\dot{M}$ as a function of $L, R, M$:

$$ \dot{M} = 